### DISCUSSION TITLE
JIT and variant order

### DISCUSSION THREAD

**Discussion by minimit** [tailwindlabs/tailwindcss#4857] [Help] [answered]
As I understand with JIT mode all variants are enabled.

So does in this file https://github.com/tailwindlabs/tailwindcss/blob/master/stubs/defaultConfig.stub.js

Is the list of variants for each property ignored `variants: {` ?
For the order of variants does JIT use `variantOrder: [` ?
In a plugin that adds a variant with `addVariant` how i specify the variant order if I don't extend `variants: {` ?

**Comment by minimit** [comment, marked as answer]
Link to updated docs https://tailwindcss.com/docs/configuring-variants#ordering-custom-variants btw with jit mode enabled it seems you cannot specify variant order

**Reply by simonswiss** [reply]
Hey!

Not sure I understand the question scope totally, but basically, in JIT mode, the `variant chains` will be generated in the order you write them in the class.

Any permutation of any variant is possible and will be generated on the fly to match the way you tried to "chain" them.

For this reason, the variant order is irrelevant and ignored.

Hope it helps!

**Reply by minimit** [reply]
Thanks
